tool/README.md

```diff
@@ -93,20 +83,46 @@ git pull upstream master
 
 #### Publish the packages
```

Comment:
I had a chicken an egg problem getting tests to run successfully in travis.  I would create a PR that had the version changes for (devtools_shared, devtools_server, and devtools_app and I guess now devtools_testing) pub those packages and commit the PR.  Then I would update the devtools pubspec commit the PR and bots would go green, then I would publish the devtools part using the publish.sh script.  So my order was (w/o the PR and commits I described above):

The order I used:
\> cd ~/devtools/packages/devtools_shared
\> pub publish
\> cd ~/devtools/packages/devtools_server
\> pub publish
\> cd ~/devtools/packages/devtools_app
\> pub publish
\> cd ~/devtools/packages/devtools_testing
\> pub publish

\> cd ~/devtools
\> ./tool/publish.sh              **// Didn't see this in this section.**

```
+ echo 'Editing .gitignore to comment out build directory'
Editing .gitignore to comment out build directory
+ pushd packages/devtools
~/devtools-git/devtools/packages/devtools ~/devtools-git/devtools
+ pub get
Resolving dependencies... 
Got dependencies!
+ rm -rf build
+ pub run webdev build

Creating build script, took 6531ms
[INFO] Reading cached asset graph completed, took 414ms
[INFO] Checking for updates since last build completed, took 677ms
[INFO] build_web_compilers|entrypoint on web/main.dart:Running dart2js with -O1 --packages=.package-3940dd1e3da738eaccfdb4b000d4b69a -oweb/main.dart.js web/main.dart
[INFO] build_web_compilers|entrypoint on web/main.dart:Dart2Js finished with:

Compiled 9,561,935 characters Dart to 2,030,648 characters JavaScript in 8.68 seconds
Dart file (web/main.dart) compiled to JavaScript: web/main.dart.js

[INFO] Running build completed, took 13.8s
[INFO] Caching finalized dependency graph completed, took 277ms
[INFO] Creating merged output dir `build` completed, took 314ms
[INFO] Writing asset manifest completed, took 3ms
[INFO] Succeeded after 14.4s with 16 outputs (1877 actions)
+ mv ./build/packages ./build/pack
+ perl -pi -e 's/^build\/$/\# build\//g' .gitignore
+ set +x
Ready to publish.

Verify the package works, then publish the package, and finally, revert the change to .gitignore.

Publish by:
cd packages/devtools
pub publish

git checkout .gitignore
~/devtools-git/devtools
```

**As the output from the publish.sh script from above stated I then publish devtools:**

\> cd packages/devtools
\> pub publish

\> git checkout .gitignore**

Then I create the tag for this release and push to the remote repository.

\> cd ~/devtools-git/devtools
\> git checkout master
\> git pull upstream master
\> git tag -a v0.0.15 -m "DevTool 0.0.15"
\> git push upstream v0.0.15

Reply:
Let me know if this makes sense.  I wish it was simpler.

Reply:
the README.md instructions say to run tool/publish.sh first before publishing the packages. Is that incorrect?

Reply:
The chicken and egg problems in travis should all go away now that the packages are version locked... assuming we haven't missed a case where the packages use version locked + local dependency override dependencies.